lua/sp.c

```diff
@@ -696,19 +706,52 @@ char *sp_column_name(struct response_data *arg, int col)
     return parent->clntname[col];
 }
 
+static int grab_qdb_table_read_lock(struct sqlclntstate *clnt,
+                                    struct dbtable *db, trigger_reg_t *reg,
+                                    int have_schema_lock)
+{
+    if (!gbl_queuedb_read_locks) return 0;
+    if ((reg != NULL) && reg->qdb_locked) {
+        return 0; /* we already have the table read lock */
+    }
+    if (!have_schema_lock && (tryrdlock_schema_lk() != 0)) {
```

Comment:
Since you are making sure to call this only once, you should now be avoiding lock ordering issues with schema-lk and the table-locks .. no?  I think you can block on the schema-lk rather than trylock.